src/traits.jl

```diff
@@ -172,6 +172,31 @@ and the easiest to use:
 """
 base_colorant_type(c::Colorant) = base_colorant_type(typeof(c))
 
+"""
+    floattype(::Type{T}) where T<:Union{Colorant, Number}
+
+Promote type `T` to AbstractFloat and strip the `Colorant` if possible.
+
+This trait is useful when concepts are only meaningful in continuous domain,
+for example, distances, gaussian distributions. It's also useful when we need
+to expand the [0,1] value-range of `FixedPointNumbers.Normed`
+
+# Examples
+
+This trait can be used together with [`base_color_type`](@ref) and [`base_colorant_type`](@ref), for example:
+
+```julia
+julia> T = RGB{N0f8}
+RGB{Normed{UInt8,8}}
+
+julia> base_colorant_type(T){floattype(T)}
+RGB{Float32}
+```
+"""
+floattype(::Type{T}) where T <: AbstractFloat = T
+floattype(::Type{T}) where T <: Bool = Float32
+floattype(::Type{T}) where T <: Colorant = floattype(eltype(T))
```

Comment:
This is *not* piracy because this package "owns" `Colorant`.

However, I think this violates the intent of `floattype`; the new type needs to be a replacement of `T` that uses floating-point arithmetic. For `RGB{N0f8}`, `RGB{Float32}` would be acceptable. Resolve this with `base_colorant_type`.

Make sure you check behavior on non-parametric types like `ARGB32`.

Reply:
perhaps what we need is `floattype_colorant` instead of `floattype` 🤔

Reply:
`floattype(eltype(c))` is shorter, clearer, and doesn't define a new name one has to go look up and read the docstring on.